akka-actor-typed-tests/src/test/scala/akka/actor/typed/scaladsl/AbstractStashSpec.scala

```diff
@@ -242,3 +243,202 @@ abstract class StashSpec extends ScalaTestWithActorTestKit with WordSpecLike {
+      probe.expectMessage("three")
+    }
+
+    "work with supervised initial Behaviors.same" in {
+      val probe = TestProbe[String]()
+      // unstashing is inside setup
+      val ref = spawn(Behaviors.supervise(Behaviors.receiveMessage[String] {
+        case "unstash" ⇒
+          val stash = StashBuffer[String](10)
+          stash.stash("one")
+          stash.unstashAll(Behavior.same)
+
+        case msg ⇒
+          probe.ref ! msg
+          Behaviors.same
+      }).onFailure[TestException](SupervisorStrategy.stop))
+
+      ref ! "unstash"
+      probe.expectMessage("one")
+      ref ! "two"
+      probe.expectMessage("two")
+    }
+
+    "work with supervised intermediate Behaviors.same" in {
+      val probe = TestProbe[String]()
+      // unstashing is inside setup
+      val ref = spawn(Behaviors.supervise(Behaviors.receiveMessage[String] {
+        case "unstash" ⇒
+          val stash = StashBuffer[String](10)
+          stash.stash("one")
+          stash.stash("two")
+          stash.unstashAll(Behaviors.receiveMessage {
+            case msg ⇒
+              probe.ref ! msg
+              Behaviors.same
+          })
+      }).onFailure[TestException](SupervisorStrategy.stop))
+
+      ref ! "unstash"
+      probe.expectMessage("one")
+      probe.expectMessage("two")
+      ref ! "three"
+      probe.expectMessage("three")
+    }
+
+    "signal PostStop to the latest unstashed behavior on failure" in {
+      val probe = TestProbe[Any]()
+      val ref =
+        spawn(stashingBehavior(probe.ref))
+
+      ref ! "stash"
+      ref ! "stash-fail"
+      ref ! "unstash"
+      probe.expectMessage("unstashing-0")
+      probe.expectMessage("stash-fail-1")
+      probe.expectMessage("post-stop-1")
+    }
+
+    "signal PostStop to the latest unstashed behavior on failure with supervision" in {
+      val probe = TestProbe[Any]()
+      val ref =
+        spawn(Behaviors.supervise(stashingBehavior(probe.ref))
+          .onFailure[TestException](SupervisorStrategy.stop))
+
+      ref ! "stash"
+      ref ! "stash-fail"
+      ref ! "unstash"
+      probe.expectMessage("unstashing-0")
+      probe.expectMessage("stash-fail-1")
+      probe.expectMessage("post-stop-1")
+    }
+
+    "signal PreRestart to the latest unstashed behavior on failure" in {
+      val probe = TestProbe[Any]()
+      val ref =
+        spawn(Behaviors.supervise(stashingBehavior(probe.ref))
+          .onFailure[TestException](SupervisorStrategy.restart))
+
+      ref ! "stash"
+      ref ! "stash-fail"
+      ref ! "unstash"
+      probe.expectMessage("unstashing-0")
+      probe.expectMessage("stash-fail-1")
+      probe.expectMessage("pre-restart-1")
+    }
+
+    "handle resume correctly on failure unstashing" in {
+      val probe = TestProbe[Any]()
+      val ref =
+        spawn(Behaviors.supervise(stashingBehavior(probe.ref))
+          .onFailure[TestException](SupervisorStrategy.resume))
+
+      ref ! "stash"
+      ref ! "stash-fail"
```

Comment:
could add one more "stash" here, could also be good if "get-current" would also reply with size of the StashBuffer, which in this case would be 1 (if we say that exceptions abort unstashing)